lib/shared/puppet_self_service.rb

```diff
@@ -121,4 +126,28 @@ def self.filesystem_free(path)
     stat = Sys::Filesystem.stat(path)
     (stat.blocks_available.to_f / stat.blocks.to_f * 100).to_i
   end
+
+  # Execute psql queries.
+  #
+  # @param sql [String] SQL to execute via a psql command.
+  # @param psql_options [String] list of options to pass to the psql command.
+  #
+  # @return (see #exec_return_result)
+  def self.psql_return_result(sql, psql_options = '')
+    command = %(su pe-postgres --shell /bin/bash --command "cd /tmp && #{PUP_PATHS[:server_bin]}/psql #{psql_options} --command \\"#{sql}\\"")
+    Facter::Core::Execution.execute(command)
+  end
+
+  # Below method to execute the PSQL statement to identify thundering herd
+  def self.psql_thundering_herd
+    sql = %(
+      SELECT count(*)
+      FROM reports
+      WHERE start_time BETWEEN now() - interval '7 days' AND now()
```

Comment:
7 days seems like a lot of data to be capturing for a test that runs every 30 minutes.